app/validators/linked_validator.rb

```diff
@@ -0,0 +1,50 @@
+class LinkedValidator < ActiveModel::EachValidator
+  DELIMITER = ':'.freeze
+
+  def validate_each(record, attribute, value)
+    register_linked = options[:register_linked]
+    record.errors.add attribute, "#{value} is not a valid CURIE" unless validate_linked(value, register_linked)
+  end
+
+  def validate_linked(value, register_linked)
+    return true if empty?(value)
+
+    return false unless valid_value?(value)
+
+    if value.include? DELIMITER
+      check_curie(value)
+    else
+      check_register(value, register_linked)
+    end
+  end
+
+  def check_curie(value)
+    values = value.split(DELIMITER)
+
+    call(values[0], values[1])
+  end
+
+  def check_register(value, register_linked)
+    return false if empty?(register_linked)
+
+    call(register_linked, value)
+  end
+
+  def call(register_name, key)
+    current_register_record = OpenRegister.record(register_name, key, :beta)
+    !current_register_record.nil?
+  rescue => e
+    return false
+  end
+
+  def valid_value?(value)
+    return true unless value.include? DELIMITER
```

Comment:
isn't this check redundant? Since we are doing the regex after that?

Reply:
The regular expression is applied is the DELIMITER is present. The check is not redundant, it's just skipping the regex or not.